test/Kestrel.Tests/DotnetTest.cs

```diff
@@ -0,0 +1,142 @@
+            }
+            finally
+            {
+                dotnetCommand.KillTree();
+            }
+        }
+
+        [Fact]
+        public void ItRunsKestrelStandaloneAfterBuild()
+        {
+            TestInstance instance = TestAssetsManager.CreateTestInstance(KestrelSampleBase)
+                                                     .WithLockFiles();
+
+            var url = NetworkHelper.GetLocalhostUrlWithFreePort();
+            var args = $"{url} {Guid.NewGuid().ToString()}";
+            var dotnetCommand = new DotnetCommand();
+            var output = Build(Path.Combine(instance.TestRoot, KestrelStandalone));
+
+            try
+            {
+                dotnetCommand.ExecuteAsync($"{output} {args}");
+                NetworkHelper.IsServerUp(url).Should().BeTrue($"Unable to connect to kestrel server - {KestrelStandalone} @ {url}");
+                NetworkHelper.TestGetRequest(url, args);
+            }
+            finally
+            {
+                dotnetCommand.KillTree();
+            }
+        }
+
+        [Fact]
+        public void ItRunsKestrelPortableAfterPublish()
+        {
+            TestInstance instance = TestAssetsManager.CreateTestInstance(KestrelSampleBase)
+                                                     .WithLockFiles();
+
+            var url = NetworkHelper.GetLocalhostUrlWithFreePort();
+            var args = $"{url} {Guid.NewGuid().ToString()}";
+            var dotnetCommand = new DotnetCommand();
+            var output = Publish(Path.Combine(instance.TestRoot, KestrelPortable), true);
+
+            try
+            {
+                dotnetCommand.ExecuteAsync($"{output} {args}");
+                NetworkHelper.IsServerUp(url).Should().BeTrue($"Unable to connect to kestrel server - {KestrelPortable} @ {url}");
+                NetworkHelper.TestGetRequest(url, args);
+            }
+            finally
+            {
+                dotnetCommand.KillTree();
+            }
+        }
+
+        [Fact]
+        public void ItRunsKestrelStandaloneAfterPublish()
+        {
+            TestInstance instance = TestAssetsManager.CreateTestInstance(KestrelSampleBase)
+                                                     .WithLockFiles();
+
+            var url = NetworkHelper.GetLocalhostUrlWithFreePort();
+            var args = $"{url} {Guid.NewGuid().ToString()}";
+            var output = Publish(Path.Combine(instance.TestRoot, KestrelStandalone), false);
+            var command = new TestCommand(output);
+
+            try
+            {
+                command.ExecuteAsync($"{args}");
+                NetworkHelper.IsServerUp(url).Should().BeTrue($"Unable to connect to kestrel server - {KestrelStandalone} @ {url}");
+                NetworkHelper.TestGetRequest(url, args);
+            }
+            finally
+            {
+                command.KillTree();
+            }
+        }
+
+        private static string Build(string testRoot)
```

Comment:
I've seen this function in a few of the test classes. We should add this to the test framework in the future.